test/index.js

```diff
@@ -1,10 +1,153 @@
 'use strict';
 
-const should = require('chai').should(); // eslint-disable-line
+require('chai').should();
 
-describe('hexo-uglifyjs', () => {
-  // Tests.
-  it('should uglify js.', () => {
-    return true;
+describe('hexo-uglify', () => {
+  const uglifyFilter = require('../lib/filter-uglifyjs');
+  const terserFilter = require('../lib/filter-terser');
+
+  it('es5 - uglify (default options)', () => {
+    const ctx = {
+      config: {
+        uglify: {
+          mangle: true,
+          output: {},
+          compress: {},
+          exclude: '*.min.js'
+        }
+      }
+    };
+
+    const code = `
+    var x = {
+        baz_: 0,
+        foo_: 1,
+        calc: function() {
+            return this.foo_ + this.baz_;
+        }
+    };
+    x.bar_ = 2;
+    x["baz_"] = 3;
+    console.log(x.calc());`;
+
+    const result = uglifyFilter.apply(ctx, [code, { path: 'source/test.js' }]);
+
+    result.should.eql('var x={baz_:0,foo_:1,calc:function(){return this.foo_+this.baz_},bar_:2,baz_:3};console.log(x.calc());');
```

Comment:
when hardcoding the result, it could break in newer version of uglify/terser (just like https://github.com/hexojs/hexo/pull/3829). I think comparing `String.length` should suffice.

Reply:
If I just use:

```js
const uglify = require('uglify-js');

uglifyFilter.apply(ctx, [code, { path: 'source/test.js' }]).should.eql(uglify(code));
```

Then the unit-test will become just useless.

Reply:
what I mean is `result.length.should.below(code.length)`.

Reply:
Even `uglifyFilter.apply(ctx, [code, { path: 'source/test.js' }]).should.eql(uglify(code));` is useful

For example, try parsing `config.uglify.priority` without `delete jsOptions.priority`, the unit test would fail. In this case, the unit test correctly identify that the plugin wouldn't work if `delete jsOptions.priority` is missing.

Reply:
> what I mean is `result.length.should.below(code.length)`.

That sounds as a good idea!

Reply:
> For example, try parsing `config.uglify.priority` without `delete jsOptions.priority`, the unit test would fail. In this case, the unit test correctly identify that the plugin wouldn't work if `delete jsOptions.priority` is missing.

If something like `delete jsOptions.priority` is missing, current unit test will not passed as well because when no valid configuration was passed to UglifyJS / Terser will resulted in `undefined`.